qiskit/pulse/channels.py

```diff
@@ -59,6 +59,57 @@ def index(self) -> int:
         """
         return self._index
 
+    def _validate_index(self, index: Any) -> None:
+        """Raise a PulseError if the channel index is invalid, namely, if it's not a positive
+        integer.
+
+        Raises:
+            PulseError: If ``index`` is not a nonnegative integer.
+        """
+        if isinstance(index, ParameterExpression) and index.parameters:
+            # Parameters are unbound
+            return
+        elif isinstance(index, ParameterExpression):
+            index = float(index)
+            if index.is_integer():
+                index = int(index)
+
+        if not isinstance(index, int) and index < 0:
+            raise PulseError('Channel index must be a nonnegative integer')
+
+    @property
+    def parameters(self) -> Set:
+        """Parameters which determine the channel index."""
+        return self._parameters
+
+    def is_parameterized(self) -> bool:
+        """Return True iff the channel is parameterized."""
+        return bool(self.parameters)
+
+    def assign(self, parameter: Parameter, value: ParameterValueType) -> 'Channel':
+        """Return a new channel with the input Parameter assigned to value.
+
+        Args:
+            parameter (Parameter): A parameter in this expression whose value will be updated.
+            value: The new value to bind to.
+
+        Returns:
+            A new channel with updated parameters.
+
+        Raises:
+            PulseError: If the parameter is not present in the channel.
+        """
+        if parameter not in self.parameters:
+            raise PulseError('Cannot bind parameters ({}) not present in the channel.'
+                             ''.format(parameter))
+
+        new_index = self.index.assign(parameter, value)
+        if not new_index.parameters:
+            self._validate_index(new_index)
+            new_index = int(new_index)
+
+        return type(self)(new_index)
```

Comment:
This returns a new channel, does this work with the other schedule parameter implementation? IIRC these mutated the schedule?

Reply:
What other implementation?